ksqldb-engine/src/main/java/io/confluent/ksql/internal/PullQueryExecutorMetrics.java

```diff
@@ -129,12 +130,21 @@ public void recordLatency(
       final PullPhysicalPlanType planType,
       final RoutingNodeType routingNodeType
   ) {
+    final MetricsKey key = new MetricsKey(sourceType, planType, routingNodeType);
+    innerRecordLatency(startTimeNanos, key);
+  }
+
+  public void recordLatencyForError(final long startTimeNanos) {
+    final MetricsKey key = new MetricsKey();
+    innerRecordLatency(startTimeNanos, key);
+  }
```

Comment:
Added these methods to make it clear when we know all the metric dimensions or none of them.